util/dvsim/sim_utils.py

```diff
@@ -99,3 +100,90 @@ def vcs_cov_summary_table(buf):
 
     # If we reached here, then we were unable to extract the coverage.
     raise SyntaxError(f"Coverage data not found in {buf.name}!")
+
+
+def get_runtimes(run_log_txt: str, tool: str) -> Tuple[int, int]:
+    """Capture and return two runtimes: job_runtime and simulated_time.
+
+    The run.log file is passed as an input to the function, in addition to the
+    tool used. The tool returns two integers if the runtimes were extracted
+    successfully. The function will raise appropriate exception if the extraction
+    fails.
+    """
+    with open(run_log_txt, 'r') as f:
+        if tool == 'xcelium':
+            return xcelium_runtimes(f)
+        elif tool == 'vcs':
+            return vcs_runtimes(f)
+        else:
+            # TODO: support using default dvsim run time
+            raise NotImplementedError(f"{tool} is unsupported for run times extraction.")
+
+
+def vcs_runtimes(buf):
```

Comment:
```suggestion
def vcs_runtimes(buf) -> Tuple[int, int]:
```

Same below.

Reply:
I am trying to search a type for `buf`. I guess we do not have one for IO wrappers.